internal/cli/execute.go

```diff
@@ -177,3 +179,32 @@ func runExecute(cmd *cobra.Command, opts executeOptions) error {
 
 	return nil
 }
+
+func makeRetryMessage(values recipe.VariableValues) string {
+	var commandline strings.Builder
+	skipNext := false
+	for idx, arg := range os.Args {
+		if skipNext {
+			skipNext = false
+			continue
+		}
+		if arg == "--set" {
+			skipNext = true
+			continue
+		}
+		// quote all non-option args to be on the safe side, except for indices 0 and 1
+		// (which are the program name and the command name)
+		if idx <= 1 || strings.HasPrefix(arg, "-") {
+			commandline.WriteString(arg)
+		} else {
+			commandline.WriteString(fmt.Sprintf("\"%s\"", arg))
+		}
+		commandline.WriteString(" ")
+	}
+
+	for key, value := range values {
+		commandline.WriteString(fmt.Sprintf(" --set \"%s=%s\"", key, value))
```

Comment:
For more complex variable types, we need to make sure that the value is serialized correctly. For example for table variables we need to serialize it to CSV format. Currently it outputs `--set "project_name=[map[1:a 2:b]]"`

Reply:
Heck, I thought my godog test tested that, but I guess it doesn't.

Reply:
I removed the test for now, I couldn't think of a way to test this in godog :(